lib/core/versions/latest/OperationProcessor.ts

```diff
@@ -103,37 +103,39 @@ export default class OperationProcessor implements IOperationProcessor {
       return didState;
     }
 
+    // When delta parsing fails, operation.delta is undefined.
     const operation = await CreateOperation.parse(anchoredOperationModel.operationBuffer);
 
+    const newDidState: DidState = {
+      document: { },
+      nextRecoveryCommitmentHash: operation.suffixData.recoveryCommitment,
+      nextUpdateCommitmentHash: undefined,
+      lastOperationTransactionNumber: anchoredOperationModel.transactionNumber
+    };
+
     // Ensure actual delta hash matches expected delta hash.
     const isMatchingDelta = Multihash.isValidHash(operation.encodedDelta, operation.suffixData.deltaHash);
     if (!isMatchingDelta) {
-      return didState;
+      return newDidState;
     }
 
     // Apply the given patches against an empty object.
     const delta = operation.delta;
     let document = { };
-    try {
-      if (delta !== undefined) {
+    if (delta !== undefined) {
+      // update the commitment hash regardless
+      newDidState.nextUpdateCommitmentHash = delta.updateCommitment;
+      try {
         document = DocumentComposer.applyPatches(document, delta.patches);
+        newDidState.document = document;
+      } catch (error) {
+        const didUniqueSuffix = anchoredOperationModel.didUniqueSuffix;
+        const transactionNumber = anchoredOperationModel.transactionNumber;
+        console.debug(
+          `Unable to apply document patch in transaction number ${transactionNumber} for DID ${didUniqueSuffix}: ${SidetreeError.stringify(error)}.`);
```

Comment:
The original debug log will sound like the entire operation is not applied. Can we update this it indicate that this is "partially applied" with the next commitment hash?